src/components/fieldset/fieldset.test.js

```diff
@@ -0,0 +1,229 @@
+import * as React from "react";
+import Fieldset from "./fieldset.component";
+import legendPreview from "../../../cypress/locators/fieldset";
+import Textbox from "../textbox/textbox.component";
+import Checkbox from "../checkbox/checkbox.component";
+import Form from "../form/form.component";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import { getDataElementByValue } from "../../../cypress/locators/index";
+import { positionOfElement } from "../../../cypress/support/helper";
+
+const specialCharacters = [
+  "legend",
+  "mp150ú¿¡üßä",
+  "!@#$%^*()_+-=~[];:.,?{}&\"'<>",
+];
+
+const FieldsetComponent = ({ ...props }) => {
+  return (
+    <div>
+      <Fieldset legend="Fieldset" {...props}>
+        <Textbox
+          label="First Name"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+        <Textbox
+          label="Last Name"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+        <Textbox
+          label="Address"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+        <Checkbox
+          label="Checkbox"
+          labelAlign="right"
+          labelWidth={30}
+          labelSpacing={2}
+          reverse
+        />
+        <Textbox label="City" labelInline labelAlign="right" labelWidth={30} />
+        <Textbox
+          label="Country"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+        <Textbox
+          label="Telephone"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+      </Fieldset>
+    </div>
+  );
+};
+
+context("Testing Fieldset component", () => {
+  describe("should render Fieldset component", () => {
+    it.each(specialCharacters)(
+      "should verify Fieldset preview text is %s",
+      (chars) => {
+        CypressMountWithProviders(<FieldsetComponent legend={chars} />);
+
+        legendPreview().should("have.text", chars);
+      }
+    );
+
+    it("should verify Fieldset preview is not displayed", () => {
+      CypressMountWithProviders(<FieldsetComponent legend="" />);
+
+      legendPreview().should("not.exist");
+    });
+
+    it.each([
+      ["inline", true, 36, 39, 78],
+      ["as a column", false, 18, 76, 919],
+    ])(
+      "should verify Fieldset is displayed %s if inline prop is %s",
+      (state, bool, labelHeight, labelWidth, inputWidth) => {
+        CypressMountWithProviders(<FieldsetComponent inline={bool} />);
+
+        const upperLabelHeight = labelHeight + 1;
+        const upperLabelWidth = labelWidth + 1;
+        const upperInputWidth = inputWidth + 1;
+
+        getDataElementByValue("label")
```

Comment:
thanks for fixing this, but this could be more simplified. Instead of using the same code 3 times, we can create a `const` like:

`
const verifyCssProps = (element, cssProp, lessValue, greaterValue) => {
getDataElementByValue(element)
           .invoke("css", cssProp)
           .then(parseFloat)
           .then(($el) => {
             expect($el).to.be.gt(lessValue);
             expect($el).to.be.lt(greaterValue);
           })
}
`

and then use it 3 times below.